moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -874,83 +877,193 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "No velocity limit was defined for joint "
+                                          << vars[idx].c_str()
+                                          << "! You have to define velocity limits in the URDF or "
+                                             "joint_limits.yaml");
+      return false;
+    }
+
+    // ACCELERATION LIMIT
+    // Check if a custom limit was specified as an argument
+    bool set_acceleration_limit = false;
+    it = acceleration_limits.find(vars[idx]);
+    if (it != acceleration_limits.end())
+    {
+      max_acceleration[idx] = it->second * acceleration_scaling_factor;
+      set_acceleration_limit = true;
+    }
+
+    if (bounds.acceleration_bounded_ && !set_acceleration_limit)
+    {
+      // Set the default acceleration limit, from robot model
+      if (bounds.max_acceleration_ < 0.0)
       {
         ROS_ERROR_NAMED(LOGNAME, "Invalid max_acceleration %f specified for '%s', must be greater than 0.0",
-                        bounds.max_acceleration_, vars[j].c_str());
+                        bounds.max_acceleration_, vars[idx].c_str());
         return false;
       }
-      max_acceleration[j] = std::min(std::fabs(bounds.max_acceleration_), std::fabs(bounds.min_acceleration_)) *
-                            acceleration_scaling_factor;
+      max_acceleration[idx] = std::min(std::fabs(bounds.max_acceleration_), std::fabs(bounds.min_acceleration_)) *
+                              acceleration_scaling_factor;
+      set_acceleration_limit = true;
+    }
+    if (!set_acceleration_limit)
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "No acceleration limit was defined for joint "
+                                          << vars[idx].c_str()
+                                          << "! You have to define acceleration limits in the URDF or "
+                                             "joint_limits.yaml");
+      return false;
     }
   }
 
+  return doTimeParameterizationCalculations(trajectory, max_velocity, max_acceleration);
+}
+
+bool TimeOptimalTrajectoryGeneration::doTimeParameterizationCalculations(robot_trajectory::RobotTrajectory& trajectory,
+                                                                         const Eigen::VectorXd& max_velocity,
+                                                                         const Eigen::VectorXd& max_acceleration) const
+{
+  // This lib does not actually work properly when angles wrap around, so we need to unwind the path first
+  trajectory.unwind();
+
+  const moveit::core::JointModelGroup* group = trajectory.getGroup();
+  if (!group)
+  {
+    ROS_ERROR_NAMED(LOGNAME, "It looks like the planner did not set the group the plan was computed for");
+    return false;
+  }
+
+  if (hasMixedJointTypes(group))
+  {
+    ROS_WARN_NAMED(LOGNAME, "There is a combination of revolute and prismatic joints in the robot model. TOTG's "
+                            "`path_tolerance` will not function correctly.");
```

Comment:
Could you please elaborate on this? I guess you refer to different units/scaling required for revolute vs. prismatic joints, while path_tolerance only provides a single parameter?

Reply:
Yeah, you've got the idea. Will tweak the comment.